docker/scripts/delete-indexed-documents

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail -u
+
+SCRIPT_NAME=$(basename "$0")
+HELP=$(cat <<EOH
+Usage: $SCRIPT_NAME --all
+   or: $SCRIPT_NAME COLLECTION... (for example: $SCRIPT_NAME artist release)
+   or: $SCRIPT_NAME --help
+
+For all or each of the given MusicBrainz Solr collections,
+delete the indexed documents.
+
+Pre-requisites:
+  Solr must be running (in SolrCloud mode).
+EOH
+)
+
+# Parse arguments
+
+if [ $# -eq 0 ]
+then
+	echo >&2 "$SCRIPT_NAME: missing argument"
+	echo >&2 "Try '$SCRIPT_NAME --help' for usage."
+	exit 64 # EX_USAGE
+elif [[ "$*" =~ ^-*h(elp)?$ ]]
+then
+	echo "$HELP"
+	exit 0 # EX_OK
+elif [[ $# -eq 1 && $1 =~ ^all$ ]]
+then
+	collections=()
+else
+	collections=("$*")
+fi
+
+# Check pre-requisites
+
+max_try=${MAX_TRY:-12}
+wait_seconds=${WAIT_SECONDS:-5}
+local_tries=1
+while [[ -f $SOLR_HOME/musicbrainz-solrcloud-first-run && $local_tries -lt $max_try ]]
+do
+	local_tries=$((local_tries + 1))
+	sleep "$wait_seconds"
+done
+if [[ -f $SOLR_HOME/musicbrainz-solrcloud-first-run && $local_tries -eq $max_try ]]
+then
+	echo >&2 "$SCRIPT_NAME: Solr first run is not complete (after waiting for $((max_try * wait_seconds)) seconds)"
+	echo >&2 "Try '$SCRIPT_NAME help' for pre-requisites."
+	exit 69 # EX_UNAVAILABLE
+fi
+if ! wait-for-solr.sh --max-attempts "$max_try" --wait-seconds "$wait_seconds"
+then
+	echo >&2 "$SCRIPT_NAME: Solr is not running (after waiting for $((max_try * wait_seconds)) seconds)"
+	echo >&2 "Try '$SCRIPT_NAME help' for pre-requisites."
+	exit 69 # EX_UNAVAILABLE
+fi
+
+# List all the collections if needed
+
+if [[ ${#collections[@]} -eq 0 ]]
+then
+	MAIN_DIR="/usr/lib/mbsssss"
+	EXCLUDE_DIRS="lib common _template"
+
+	# List subdirectories
+	SUBDIRS=$(find $MAIN_DIR -maxdepth 1 -mindepth 1 -type d -print0 | xargs -0 -n 1 basename | sort)
+
+	# Exclude specified directories
+	if [ -n "$EXCLUDE_DIRS" ]
```

Comment:
Can this be empty given the assignment above? It seems to be always set.

Reply:
It can’t be empty indeed. This is legacy code I copied from the script `create-musicbrainz-collections` where `EXCLUDE_DIRS` was defined under _Parameters_.

I simplified the code here to pass `-not -name` arguments to the command `find` instead, through the amending commit https://github.com/metabrainz/mb-solr/pull/61/commits/425e8261a8a54865bba5be414be264f7653b94ef.